theano/sandbox/cuda/dnn.py

```diff
@@ -368,6 +368,27 @@ def ensure_float(val, default, name):
     return val
 
 
+def do_merge_gpu_alloc_empty(node1, node2):
```

Comment:
We are checking if two GpuDnnConv have all the same inputs except for the third one (the output buffer) where it must either be the same or be two GpuAllocEmpty.